move_group/src/move_group_context.cpp

```diff
@@ -89,3 +89,41 @@ bool move_group::MoveGroupContext::status() const
     return false;
   }
 }
+
+/** Validate that first point of given a trajectory matches current state of robot */
+bool move_group::MoveGroupContext::validateTrajectory(const trajectory_msgs::JointTrajectory &trajectory) const
+{
+  if (trajectory.points.empty())
+    return true;
+
+  const planning_scene_monitor::CurrentStateMonitorPtr csm = planning_scene_monitor_->getStateMonitor();
+  robot_state::RobotStatePtr current_state;
+  if (!csm->waitForCurrentState(ros::Time::now()) || !(current_state = csm->getCurrentState()))
+  {
+    ROS_WARN("Failed to receive full current joint state");
```

Comment:
Please make that a `ROS_ERROR` (the failure cases below are).
Also please add something like "Failed to verify the proposed trajectory is valid" to the log output.
This should make the problem more apparent.